scripts/deploy-federation.sh

```diff
@@ -84,7 +84,16 @@ shift
 JOIN_CLUSTERS="${*}"
 
 if [[ ! "${USE_LATEST}" ]]; then
-  docker build . -f Dockerfile.controller -t "${IMAGE_NAME}"
+  base_dir="$(cd "$(dirname "$0")/.." ; pwd)"
+  dockerfile_dir="${base_dir}/images/federation-v2"
+  if [[ ! -f "$base_dir/bin/controller-manager" && ! -f "${dockerfile_dir}/controller-manager" ]] ; then
+    echo "$base_dir/bin/controller-manager not found, building"
+    go build -o "${dockerfile_dir}"/controller-manager "${base_dir}"/cmd/controller-manager/main.go
```

Comment:
I think it's preferable to use the Makefile now that it's available instead of running these commands explicitly.